### ISSUE TITLE
@contextmanager disables ParamSpec

### ISSUE THREAD

**Issue by bukzor-sentryio** [microsoft/pyright#6741]
**Describe the bug**

pyright seems to stop treating paramspecs as typevars when wrapped in a @contextmanager.

**Code or Screenshots**

```python
from contextlib import contextmanager
from typing import Callable, Generator, ParamSpec

P = ParamSpec("P")


@contextmanager
def called(
    func: Callable[P, str],
    *func_args: P.args,
    **func_kwargs: P.kwargs,
) -> Generator[str, None, None]:
    yield func(*func_args, **func_kwargs)


def myfunc(s: int, t: float) -> str:
    return "hi!"


with called(myfunc, 3, 1.1) as s:
    print(s)


__demo_output__ = """
$ pyright --version
pyright 1.1.337

$ pyright demo.py
./demo.py
  ./demo.py:20:21 - error: Expected 1 positional argument (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations

$ mypy --version
mypy 1.7.1 (compiled: yes)

$ mypy demo.py
Success: no issues found in 1 source file
"""


```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release. I've included a variant of your code sample in the pyright unit tests since we didn't have this variant covered previously.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.341, which I just published. It will also be included in a future release of pylance.
